File: ProgressNotification/app/src/main/java/com/akexorcist/progressnotification/data/ApplyPremiumAccountResult.java
package com.akexorcist.progressnotification.data;

import android.os.Parcel;
import android.os.Parcelable;
/**
 * Created by Akexorcist on 8/27/2017 AD.
 */

public class ApplyPremiumAccountResult implements Parcelable {
    private boolean isActive;

    public ApplyPremiumAccountResult(boolean isActive) {
        this.isActive = isActive;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeByte((byte) (isActive ? 1 : 0));
    }

    protected ApplyPremiumAccountResult(Parcel in) {
        isActive = in.readByte() != 0;
    }

    public static final Creator<ApplyPremiumAccountResult> CREATOR = new Creator<ApplyPremiumAccountResult>() {
        @Override
        public ApplyPremiumAccountResult createFromParcel(Parcel in) {
            return new ApplyPremiumAccountResult(in);
        }

        @Override
        public ApplyPremiumAccountResult[] newArray(int size) {
            return new ApplyPremiumAccountResult[size];
        }
    };
}

File: app/src/main/java/com/akexorcist/progressnotification/data/ApplyPremiumAccountResult.kt
package com.akexorcist.progressnotification.data

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Akexorcist on 8/27/2017 AD.
 */

class ApplyPremiumAccountResult(var isActive: Boolean = false) : Parcelable {
    constructor(parcel: Parcel) : this(parcel.readByte() != 0.toByte())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeByte(if (isActive) 1 else 0)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ApplyPremiumAccountResult> {
        override fun createFromParcel(parcel: Parcel): ApplyPremiumAccountResult {
            return ApplyPremiumAccountResult(parcel)
        }

        override fun newArray(size: Int): Array<ApplyPremiumAccountResult?> {
            return arrayOfNulls(size)
        }
    }
}